react/game/src/app/screens/Login/index.js

```diff
@@ -1,44 +1,44 @@
 import React, { Component, Fragment } from 'react';
-import { withRouter, Redirect } from 'react-router-dom';
+import { withRouter } from 'react-router-dom';
 import { connect } from 'react-redux';
-import PropTypes from 'prop-types';
+import { compose } from 'redux';
 
 import loginActions from '../../../redux/auth/actions';
-import LIST from '../../../constants/routes';
+import withLoading from '../../components/WithLoading';
 
 import LoginForm from './components/loginForm';
+import withAuth from './components/withAuth';
 
 class LoginFormContainer extends Component {
-  setRedirect = values => {
+  setRedirect = async values => {
     const { dispatch } = this.props;
-    dispatch(loginActions.login(values));
+    await dispatch(loginActions.loaderActive());
```

Comment:
Remove this action here. The layout shouldn't handle this reducer logic. U must only dispatch `dispatch(loginActions.login(values));`. Also, handle the `setTimeout` in the action, not here